backend/src/main/java/com/festago/festival/application/FestivalSearchV1QueryService.java

```diff
@@ -15,16 +24,64 @@
 @RequiredArgsConstructor
 public class FestivalSearchV1QueryService {
 
-    private static final Pattern SCHOOL_PATTERN = Pattern.compile(".*대(학교)?$");
-
     private final ArtistFestivalSearchV1QueryDslRepository artistFestivalSearchV1QueryDslRepository;
     private final SchoolFestivalSearchV1QueryDslRepository schoolFestivalSearchV1QueryDslRepository;
 
     public List<FestivalSearchV1Response> search(String keyword) {
-        Matcher schoolMatcher = SCHOOL_PATTERN.matcher(keyword);
-        if (schoolMatcher.matches()) {
-            return schoolFestivalSearchV1QueryDslRepository.executeSearch(keyword);
+        return sortFestival(findFestivals(keyword));
+    }
+
+    private List<FestivalSearchV1Response> findFestivals(String keyword) {
+        if (artistFestivalSearchV1QueryDslRepository.existsByName(keyword)) {
+            return artistFestivalSearchV1QueryDslRepository.executeSearch(keyword);
+        }
+        return schoolFestivalSearchV1QueryDslRepository.executeSearch(keyword);
+    }
+
+    private List<FestivalSearchV1Response> sortFestival(List<FestivalSearchV1Response> festivals) {
+        Map<FestivalFilter, List<FestivalSearchV1Response>> festivalByStatus = divideByStatus(festivals);
+        List<FestivalSearchV1Response> result = new ArrayList<>();
+        for (FestivalFilter festivalFilter : determineFestivalOrder()) {
+            List<FestivalSearchV1Response> festivalsByFilter = festivalByStatus.getOrDefault(festivalFilter,
+                new ArrayList<>());
+            sortByStatus(festivalFilter, festivalsByFilter);
+            result.addAll(festivalsByFilter);
+        }
+        return result;
+    }
+
+    private Map<FestivalFilter, List<FestivalSearchV1Response>> divideByStatus(
+        List<FestivalSearchV1Response> festivals) {
+        return festivals.stream()
+            .collect(Collectors.groupingBy(
+                this::determineStatus,
+                () -> new EnumMap<>(FestivalFilter.class),
+                Collectors.toList()
+            ));
+    }
```

Comment:
줄바꿈을 다음과 같이 사용하는건 어떤가요?
```suggestion
    private Map<FestivalFilter, List<FestivalSearchV1Response>> divideByStatus(
        List<FestivalSearchV1Response> festivals
    ) {
        return festivals.stream()
            .collect(Collectors.groupingBy(
                this::determineStatus,
                () -> new EnumMap<>(FestivalFilter.class),
                Collectors.toList()
            ));
    }
```